R-package/R/lgb.Booster.R

```diff
@@ -790,12 +788,23 @@ Booster <- R6::R6Class(
 #'               the values in \code{params} take precedence.
 #' @param ... ignored
 #' @return For prediction types that are meant to always return one output per observation (e.g. when predicting
-#'         \code{type="response"} on a binary classification or regression objective), will return a vector with one
-#'         element per row in \code{newdata}.
+#'         \code{type="response"} or \code{type="raw"} on a binary classification or regression objective), will
+#'         return a vector with one element per row in \code{newdata}.
 #'
 #'         For prediction types that are meant to return more than one output per observation (e.g. when predicting
-#'         \code{type="response"} on a multi-class objective, or when predicting \code{type="leaf"}, regardless of
-#'         objective), will return a matrix with one row per observation in \code{newdata} and one column per output.
+#'         \code{type="response"} or \code{type="raw"} on a multi-class objective, or when predicting \code{type="leaf"},
+#'         regardless of objective), will return a matrix with one row per observation in \code{newdata} and one column per output.
+#'
+#'         For \code{type="leaf"} predictions, will return a matrix with one row per observation in \code{newdata}
+#'         and one column per tree. Note that for multiclass objectives, LightGBM trains one tree per class. That means
+#'         that, for example, for a multiclass model with 3 classes, the leaf predictions for the first class can be found
+#'         in columns 1, 4, 7, 10, etc.
+#'
+#'         For \code{type="contrib"}, will return a matrix of SHAP values with one row per observation in \code{newdata}
+#'         and columns corresponding to features. For regression, ranking, and binary classification objectives, this matrix
+#'         contains one column per feature plus a final column containing the Shapley base value. For multiclass objectives,
```

Comment:
LightGBM also has [`cross-entropy application`](https://lightgbm.readthedocs.io/en/v3.3.2/Parameters.html#objective). Should it be mentioned here as well?

Reply:
oh right! I forget about that sometimes. Yes, probably good to include that here for completeness.

Reply:
pushed these changes in https://github.com/microsoft/LightGBM/pull/5384/commits/b3eb3205086b790dedcba9d4a89057fdc5032220

(did that instead of accepting them in the browser since the roxygen docs needed to be regenerated)